omnipaxos/src/utils/mod.rs

```diff
@@ -1,2 +1,6 @@
 /// Holds helpful functions used in creating loggers.
+#[cfg(feature = "logging")]
 pub mod logger;
+/// Holds code for ui implementation.
+#[cfg(feature = "ui")]
+pub mod ui;
```

Comment:
I would like `omnipaxos` to only hold the core logic/library, so I suggest moving this ui into its own workspace instead e.g., `omnipaxos_ui`. It would be more clean and you can simply use it as a dependency for the UI feature.